florist/api/routes/server/job.py

```diff
@@ -99,3 +100,73 @@ async def change_job_status(job_id: str, status: JobStatus, request: Request) ->
         return JSONResponse(content={"error": str(assertion_e)}, status_code=400)
     except Exception as general_e:
         return JSONResponse(content={"error": str(general_e)}, status_code=500)
+
+
+@router.get("/get_server_log/{job_id}")
+async def get_server_log(job_id: str, request: Request) -> JSONResponse:
+    """
+    Return the contents of the server's log file for the given job id.
+
+    :param job_id: (str) the ID of the job to get the server logs for.
+    :param request: (fastapi.Request) the FastAPI request object.
+
+    :return: (JSONResponse) if successful, returns the contents of the file as a string.
+        If not successful, returns the appropriate error code with a JSON with the format below:
+            {"error": <error message>}
+    """
+    try:
+        job = await Job.find_by_id(job_id, request.app.database)
+
+        assert job is not None, f"Job {job_id} not found"
+        assert (
+            job.server_log_file_path is not None and job.server_log_file_path != ""
+        ), "Log file path is None or empty"
+
+        with open(job.server_log_file_path, "r") as f:
+            content = f.read()
+            return JSONResponse(content)
+
+    except AssertionError as assertion_e:
+        return JSONResponse(content={"error": str(assertion_e)}, status_code=400)
+    except Exception as general_e:
+        return JSONResponse(content={"error": str(general_e)}, status_code=500)
+
+
+@router.get("/get_client_log/{job_id}/{client_index}")
+async def get_client_log(job_id: str, client_index: int, request: Request) -> JSONResponse:
+    """
+    Return the contents of the log file for the client with given index under given job id.
+
+    :param job_id: (str) the ID of the job to get the client logs for.
+    :param client_index: (int) the index of the client within the job.
+    :param request: (fastapi.Request) the FastAPI request object.
+
+    :return: (JSONResponse) if successful, returns the contents of the file as a string.
+        If not successful, returns the appropriate error code with a JSON with the format below:
+            {"error": <error message>}
+    """
+    try:
+        job = await Job.find_by_id(job_id, request.app.database)
+
+        assert job is not None, f"Job {job_id} not found"
+        assert job.clients_info is not None, "Job has no clients."
+        assert (
+            0 <= client_index < len(job.clients_info)
+        ), f"Client index {client_index} is invalid (total: {len(job.clients_info)})"
+
+        client_info = job.clients_info[client_index]
```

Comment:
This is almost certainly a naive question, but is there a reason we are relying on the server to tell the clients where their log files are? In my head, rather than passing the location of the log to the client, the client will have stored that path already and you would just hit it with a get call to tell it that you want its log file. There's likely something more complicated there that I'm not seeing though.

Reply:
I sort of see where the complexity is coming from. Basically you want the server to set the log location for each client. Perhaps that's something we can communicate on startup and then don't have to worry about the server managing that from there on out?